src/guiguts/widgets.py

```diff
@@ -107,6 +116,39 @@ def show_dialog(cls, dlg_cls: type[TlDlg], title: Optional[str] = None) -> TlDlg
                 ToplevelDialog._toplevel_dialogs[dlg_name] = dlg_cls()  # type: ignore[call-arg]
         return ToplevelDialog._toplevel_dialogs[dlg_name]  # type: ignore[return-value]
 
+    def _do_config(self) -> None:
+        config_dict = preferences.get("DialogGeometry")
+        try:
+            geometry = config_dict[self.__class__.__name__]
+            x_resize, y_resize = self.resizable()
+            if not (x_resize or y_resize):
+                geometry = re.sub(r"^\d+x\d+", "", geometry)
+            self.geometry(geometry)
+        except KeyError:
```

Comment:
I think it's generally better to handle known conditions directly via a conditional than with an exception -- or to have the exception narrowly focused -- to prevent something _else_ in the try block raising the exception making hard-to-track-down bugs.

Reply:
Thanks @cpeel - is this better? Still trying to learn the balance/benefits between EAFP & LBYL styles.